torch/nn/modules/upsampling.py

```diff
@@ -48,9 +48,12 @@ class Upsample(Module):
         0.3.1. Since then, the default behavior is ``align_corners = False``.
         See below for concrete examples on how this affects the outputs.
 
+    .. warning::
+        This class is deprecated in favor of :func:`~nn.functional.interpolate`.
```

Comment:
I have mixed feelings about deprecating `nn.Upsample` without giving a module alternative to the users.
In simple cases(like interpolating by a fixed scale factor), it might still be handy to provide a module.
Let's see what the others think.